trustd/src/openapi.rs

```diff
@@ -0,0 +1,57 @@
+use std::fs;
+use std::path::PathBuf;
+use std::process::ExitCode;
+
+use anyhow::{anyhow, Error, Result};
+
+use trustify_server::openapi::openapi;
+
+#[derive(clap::Args, Debug)]
+pub struct Run {
+    #[command(subcommand)]
+    pub(crate) command: Command,
+}
+
+#[derive(clap::Subcommand, Debug)]
+pub enum Command {
+    Export(Export),
+}
+
+impl Run {
+    pub async fn run(self) -> Result<ExitCode> {
+        use Command::*;
+        match self.command {
+            Export(export) => export.run().await,
+        }
+    }
+}
+
+#[derive(clap::Args, Debug)]
+pub struct Export {
+    /// The file the openapi spec should be exported to
+    #[arg(long, env)]
+    pub file: PathBuf,
+}
+
+impl Export {
+    pub async fn run(self) -> Result<ExitCode> {
+        let doc = match self.file.file_name() {
```

Comment:
Works the same, just pointing it out. I guess you could also use `std::fs::PathBuf::extension` to get the extension and then do a match: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=30e4f2062ea0893781d7ad2845cd971a